romancal/skycell/skymap.py

```diff
@@ -0,0 +1,724 @@
+    """
+    pixelshift = models.Shift(-wcsinfo["x0_projection"], name="crpix1") & models.Shift(
+        -wcsinfo["y0_projection"], name="crpix2"
+    )
+    pixelscale = models.Scale(wcsinfo["pixel_scale"], name="cdelt1") & models.Scale(
+        wcsinfo["pixel_scale"], name="cdelt2"
+    )
+    tangent_projection = models.Pix2Sky_TAN()
+    celestial_rotation = models.RotateNative2Celestial(
+        wcsinfo["ra_projection_center"], wcsinfo["dec_projection_center"], 180.0
+    )
+
+    matrix = wcsinfo.get("rotation_matrix", None)
+    if matrix is not None:
+        matrix = np.array(matrix)
+    else:
+        matrix = np.reshape(
+            wcs_util.calc_rotation_matrix(
+                np.deg2rad(wcsinfo.get("orientat_projection_center", 0.0)),
+                v3i_yangle=0.0,
+                vparity=1,
+            ),
+            (2, 2),
+        )
+    rotation = models.AffineTransformation2D(matrix, name="pc_rotation_matrix")
+    det2sky = (
+        pixelshift | rotation | pixelscale | tangent_projection | celestial_rotation
+    )
+    det2sky.name = "linear_transform"
+
+    detector_frame = coordinate_frames.Frame2D(
+        name="detector", axes_names=("x", "y"), unit=(u.pix, u.pix)
+    )
+    sky_frame = coordinate_frames.CelestialFrame(
+        reference_frame=coordinates.ICRS(), name="icrs", unit=(u.deg, u.deg)
+    )
+    wcsobj = WCS(
+        [(detector_frame, det2sky), (sky_frame, None)], name=wcsinfo.get("name", None)
+    )
+
+    if bounding_box:
+        wcsobj.bounding_box = bounding_box
+
+    return wcsobj
+
+
+def skycell_to_wcs(skycell_record: dict) -> WCS:
+    """From a skycell record, generate a WCS
+
+    Parameters
+    ----------
+    skycell_record : dict
+        A skycell record, or row, from the skycell patches table.
+
+    Returns
+    -------
+    wcsobj : WCS
+        The WCS object from the skycell record.
+    """
+    wcsinfo = dict()
+
+    # The scale is given in arcseconds per pixel. Convert to degrees.
+    wcsinfo["pixel_scale"] = float(skycell_record["pixel_scale"]) / 3600.0
+
+    # Remaining components of the wcsinfo block
+    wcsinfo["ra_ref"] = float(skycell_record["ra_projection_center"])
+    wcsinfo["dec_ref"] = float(skycell_record["dec_projection_center"])
+    wcsinfo["x_ref"] = float(skycell_record["x0_projection"])
+    wcsinfo["y_ref"] = float(skycell_record["y0_projection"])
+    wcsinfo["orientat"] = float(skycell_record["orientat_projection_center"])
+
+    # Bounding box of the skycell. Note that the center of the pixels are at (0.5, 0.5)
+    bounding_box = (
+        (-0.5, -0.5 + skycell_record["nx"]),
+        (-0.5, -0.5 + skycell_record["ny"]),
+    )
+
+    wcsobj = wcsinfo_to_wcs(wcsinfo, bounding_box=bounding_box)
+
+    wcsobj.array_shape = tuple(
+        int(axs[1] - axs[0] + 0.5)
+        for axs in wcsobj.bounding_box.bounding_box(order="C")
+    )
+    return wcsobj
+
+
+def to_skycell_wcs(library: ModelLibrary) -> WCS | None:
```

Comment:
Again, it's not necessary for a user to know how to construct a ModelLibrary when only the association is needed.
I suggest this is changed to accept an association (file)

Reply:
It also needs a test